### DISCUSSION TITLE
Langchain-Chatchat v0.3.0

### DISCUSSION THREAD

**Discussion by imClumsyPanda** [chatchat-space/Langchain-Chatchat#4242] [Announcements]
## 重要提示
Langchain-Chatchat v0.3.0 相比 0.2.x 有较大项目结构变动，但知识库的源文件与向量库仍可与前序版本通用，建议参考项目 README.md 中 `快速上手` 章节内容进行新版本部署。

------
## 新增功能
* Langchain-Chatchat Python 库现已发布至 Pypi，可通过 pip install langchain-chatchat 方式直接安装； 
* 替换原有 FastChat 模型推理框架，支持 Xinference、Ollama、One API 等多种模型推理与在线 API 框架的接入；
* 所有 Chat 接口修改为与 OpenAI API 形式对齐，真正实现 OpenAI API In, OpenAI API Out；
* 重构 Agent 功能，针对 GLM 和 Qwen 模型进行适配优化；
* 新增 ArXiv 文献对话、Wolfram 对话、文生图等多种 Agent 可调用工具；
* 更新知识库问答功能，支持 Ensemble Retriever 接入，并将在后续接入更多不同类型 Retriever；
* 更新 Streamlit WebUI，支持模型推理框架选择、对话管理、工具选择等多种新功能；
* 增加 Text2SQL 工具，支持 SQLAlchemy 大部分数据库，支持特定表、智能判定表与 Read-Only 模式，支持对表名进行额外说明；
* 新增 chatchat、chatchat-config、chatchat-kb 命令，支持通过命令行进行项目配置、知识库重构与项目启动；
* 更多细节更新，期待你的探索发现！

## 新晋贡献者
* @yhfgyyf made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2630
* @HALIndex made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2591
* @596192804 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2013
* @Lijia0 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2480
* @xyhshen made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2648
* @MSZheng20 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2688
* @showmecodett made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2740
* @songpb made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2816
* @fengyaojieTTT made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2919
* @criwits made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2949
* @wusongbai139 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2960
* @thinklover made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/2892
* @hollowdjj made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3155
* @dignfei made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3276
* @Sumkor made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3324
* @oslijunw made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3404
* @liudichen made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3407
* @srszzw made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3508
* @RoceoRobin made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3585
* @yuehua-s made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3664
* @LOGIC-10 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3108
* @saliven1970 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3458
* @dengpeng777 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3574
* @FJiangArthur made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3340
* @Kudoryafuka3 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3804
* @An0nymous0 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3564
* @pearjelly made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3912
* @qqlww1987 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3203
* @vegetablest made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/4048
* @selfuppen made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3873
* @BestLemoon made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3887
* @liuchuan01 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3206
* @momomobinx made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3477
* @gzYou made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3821
* @klaus-xiong made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3926
* @weank1984 made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/4135
* @zhenlong-zhu made their first contribution in https://github.com/chatchat-space/Langchain-Chatchat/pull/3229

**Full Changelog**: https://github.com/chatchat-space/Langchain-Chatchat/compare/v0.2.10...v0.3.0

<hr /><em>This discussion was created from the release <a href='https://github.com/chatchat-space/Langchain-Chatchat/releases/tag/v0.3.0'>Langchain-Chatchat v0.3.0</a>.</em>

**Comment by qqlww1987** [comment]
终于来了撒花

**Comment by keithxanshang** [comment]
问下v0.3.0的源代码在哪个分支?

**Reply by qqlww1987** [reply]
master 已经正式发布了

**Comment by whpen** [comment]
撒花
